LandSandBoat/server
# Additional effect global script corrections

**_I affirm:_**

## What does this pull request do?
A prior change causes the resistance calculations to deal the damage immediately where this did not happen before, meaning when we hit the defender with the hp removal, it'd take the damage twice.

There is likely now also an issue with non-hp drains also dealing damage, but that will be handled in a minor refactor already in progress. I've also correct a mistake where it calls `math.random()` more than needed resulting in reduced drain proc rates (I dunno how I missed that)

This PR also contains a minor change in the procType enumeration and updates affected SQL rows.

## Steps to test these changes
1. BEFORE pulling these commits, equip a drain weapon, modify proc rate to be 100% to save yourself some time.
2. use GM command to take off some of your own HP so you get the full drain effect
3. print mobs HP before and after a proc, note that the damage+drain in the log is less than HP removed from the mob.
4. NOW pull down these commits and repeat steps 1 through 3, note that the damage+drain in the log actually matches the HP removed from the mob this time.
